Add documentation about the `codeQL.queryHistory.ttl` setting (#11073)
github/codeql: docs/codeql/codeql-for-visual-studio-code/customizing-settings.rst

To override the default label, you can specify a different format for the query history items.

Changing the retention period for query history items
-----------------------------------------------------

> By default, items in the query history view are retained for 30 days. This value is customizable by changing the "Code QL > Query History: TTL" setting. To retain items indefinitely, set the value to 0.

Reviewer: Suggest that we make this a little more active and introduce the "TTL" acronym:

```suggestion
By default, items in the query history view are retained for 30 days. You can set a different time to live (TTL) by changing the "Code QL > Query History: TTL" setting. To retain items indefinitely, set the value to 0.
```
